skycatalogs/objects/snana_object.py

```diff
@@ -0,0 +1,205 @@
+            return np.exp(-0.921034037196184 * m)
+        def _to_mag(f):
+            return np.log(flux)/(-0.921034037196184)
+
+        flux = super().get_LSST_flux(bandpass, sed, mjd)
+
+        mjd_ix_l, mjd_ix_u, mjd_fraction = self._find_mjd_interval(mjd)
+        if flux > 0.0:
+            with h5py.File(self._belongs_to._SED_file, 'r') as f:
+                cors = f[self._id][f'magcor_{bandpass}']
+
+                # interpolate corrections
+                if mjd_ix_l == mjd_ix_u:
+                    mag_cor = cors[mjd_ix_l]
+                else:
+                    mag_cor = cors[mjd_ix_l] + mjd_fraction *\
+                        (cors[mjd_ix_u] - cors[mjd_ix_l])
+
+                #dbg = True
+                dbg = False
+
+                # Do everything in flux units
+                flux_cor = _to_flux(mag_cor)
+                corrected_flux = flux * flux_cor
+
+                if dbg:
+                    print(f'Band {bandpass} uncorrected flux: {flux}')
+                    print(f'                mag correction: {mag_cor}')
+                    print(f' multiplicative flux correction: {flux_cor}')
+
+                if cache:
+                    att = f'lsst_flux_{band}'
+                    setattr(self, att, corrected_flux)
+        return corrected_flux
+
+    def _find_mjd_interval(self, mjd=None):
+        '''
+        Find indices into mjd array of elements bounding our mjd
+        Also compute and constant needed for interpolation.  If we're
+        using "standard" mjd, also store these numbers.
+
+        Parameters
+        ----------
+        mjd     float   If None use the one stored in our ObjectCollection
+
+        Returns
+        -------
+        A triple:     index below, index above, and fraction of distance
+                      mjd is from entry below to entry above
+        '''
+        if not mjd:
+            mjd = self._belongs_to._mjd
+            store = True
+        else:
+            store = (mjd == self._belongs_to._mjd)
+
+        if store:
+            if self._mjd_ix_l:
+                # just return previously-computed values
+                return self._mjd_ix_l, self._mjd_ix_u, self._mjd_fraction
+
+        if not self._mjds:
+            with h5py.File(self._belongs_to._SED_file, 'r') as f:
+                self._mjds = np.array(f[self._id]['mjd'])
+
+        last_ix = len(self._mjds) - 1
+        if mjd <= self._mjds[0]:
+            mjd_ix_l = 0
+            mjd_ix_u = 0
+            mjd_fraction = None
+        elif mjd >= self._mjds[last_ix]:
+            mjd_ix_l = last_ix
+            mjd_ix_u = last_ix
+            mjd_fraction = None
+        else:
+            mjd_ix_u = np.argmin((mjd - self._mjds ) > 0)
+            mjd_ix_l = mjd_ix_u - 1
+            mjds = self._mjds
+            # Where is mjd relative to mjds on either side in the array?
+            mjd_fraction = (mjd - mjds[mjd_ix_l]) /\
+                (mjds[mjd_ix_u] - mjds[mjd_ix_l])
```

Comment:
I don't have strong objections to using bisect, but don't see a clear advantage either.  The boundary cases can be checked easily without any need for a function call.  Otherwise - is there reason to believe bisect is more efficient than the code I have using `np.argmin` ?

Reply:
Based on the documentation, `np.argmin` must be an O(N) algorithm, whereas `bisect.bisect` is O(log N).  My main reason for using `bisect.bisect` would be that it seems more idiomatic to use for this kind of search of an ordered sequence.